### PR TITLE
Fix and refactor loadFile

### PR DESCRIPTION
**Pull request by ebekebe**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -96,79 +96,67 @@ module.exports = class Config {
...
     const {
       absolutePath,
       relativePath,
       isRequired = false,
       ignoreIsDirectoryError = false,
     } = options
-    let filePath = null
+    const filePath = this.resolveFilePath({absolutePath, relativePath})
+    const fileContent = this.getFileContent({filePath, isRequired})
+    const configObject = this.getConfigObject({fileContent, filePath})
```
Comment (adius) [inline index.js]: Shouldn't the methods be private?

File: index.js [hunk 2]
```diff
@@ -180,11 +174,28 @@ module.exports = class Config {
...
+  getFileContent ({filePath, isRequired}) {
+    try {
+      return fs.readFileSync(filePath)
+    }
+    catch (error) {
+      const notExistant = error.message.includes('no such file')
+      const isDirectory = error.message
+        .includes('illegal operation on a directory')
+
+      if (isRequired && (ignoreIsDirectoryError || !isDirectory)) {
```
Comment (adius) [inline index.js]: This looks wrong

File: tests/main.js [hunk 3]
```diff
@@ -83,6 +84,18 @@ const Config = require('..')
...
 
 {
+  console.info('- It calls onSuccess hook')
+  const testConfig = new Config()
+  let hookFilePath
+  testConfig.loadFile({
+    relativePath: '.no-default-name.yaml',
+    onSuccess: filePath => { hookFilePath = filePath },
+  })
+  expect(hookFilePath, 'to equal', path.resolve('.no-default-name.yaml'))
```
Comment (adius) [inline tests/main.js]: Move the check into the `onSuccess` function!?

**Review by adius — CHANGES_REQUESTED** [review]

File: tests/main.js (hunk 3 above)
Comment (ebekebe) [reply tests/main.js]: Needs to stay outside, otherwise the test will always succeed.

**Review by adius — APPROVED** [review]

**Comment by adius** [comment]
I guess you can squash it

**Pull request merged**